src/server/locals.js

```diff
@@ -135,35 +151,102 @@ function getLocalScopeForFrame(matchedFrame, callback) {
 
     var locals = response.result;
     matchedFrame.stackLocation.locals = {};
-    for (var local of locals) {
-      matchedFrame.stackLocation.locals[local.name] = getLocalValue(local);
+    var localsContext = {
+      localsObject: matchedFrame.stackLocation.locals,
+      options: options,
+      depth: options.depth
     }
-
-    callback(null);
+    async.each(locals, getLocalValue.bind(localsContext), callback);
   });
 }
 
-function getLocalValue(local) {
-  var value;
+function getLocalValue(local, callback) {
+  var localsObject = this.localsObject;
+  var options = this.options;
+  var depth = this.depth;
+
+  function cb(error, value) {
+    if (error) {
+      // Add the relevant data to the error object,
+      // taking care to preserve the innermost data context.
+      if (!error.rollbarContext) {
+        error.rollbarContext = local;
+      }
+      return callback(error);
+    }
 
-  switch (local.value.type) {
-    case 'undefined': value = 'undefined'; break;
-    case 'object': value = getObjectValue(local); break;
-    case 'array': value = getObjectValue(local); break;
-    default: value = local.value.value; break;
+    if (_.typeName(localsObject) === 'array') {
+      localsObject.push(value);
+    } else {
+      localsObject[local.name] = value;
+    }
+    callback(null);
   }
 
-  return value;
+  if(!local.value) {
```

Comment:
Nit: there's a mix of styles in the PR. Some are `if(` while others are `if (`, the latter seeming to dominate. Ideally, all would have the same style.